### DISCUSSION TITLE
AsciiDoc Linter

### DISCUSSION THREAD

**Discussion by FelixGeisler** [docToolchain/docToolchain#739] [Q&A] [answered]
Hey,

are there any plans to add linters to the docToolchain? 
Due to the docToolchain we are considering switching from markdown to AsciiDoc. Does anybody know a good AsciiDoc linter?

Greetings
Felix

**Comment by rdmueller** [comment, marked as answer]
thanx for asking. 

What do you expect from such a linter? I've seen tools for markdown which check the syntax and even grammar and readability.

I like those tools, however, I am not aware of an asciidoc specific linter. 

What I use for my projects is htmlSanityCheck to check for broken links etc. https://aim42.github.io/htmlSanityCheck/report/report.html

And I also use the IntelliJ plugin which provides grammar checks and also some Asciidoctor warnings within the IDE.

btw: have you already seen the multi-markup-demo? http://doctoolchain.org/multi-markup-demo/demo/markdown.html 
This might help you in the transition from markdown to AsciiDoc. We are working on an even better integration of several markup languages.

**Reply by FelixGeisler** [reply]
Thanks for the fast reply.

We use linters to implement a consistent code style and formatting in our architecture and documentation. For markdown we use markdownlint. I was wondering if there might be an alternative for AsciiDoc, but i couldnt find one either.

The htmlSanityCheck is a nice tool that we will definitly use. 

We are currently using VS Code with the AsciiDoc extension. It provides nice features like syntax highlighting and a live preview. 

Thank you for the multi-markup-demo. This will help us with the first steps.

**Reply by rdmueller** [reply]
hm. I like the ruleset defined by markdownlint: https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint
Maybe we can take this ruleset and transfer it to the AsciiDoc-World. I'll open up an issue for this.

**Reply by rdmueller** [reply]
https://github.com/docToolchain/docToolchain/issues/741

**Reply by FelixGeisler** [reply]
Thank you for creating the issue.

Unfortunately, I don't have time to implement this myself at the moment. But I will have a look at it in a few weeks.

**Reply by rdmueller** [reply]
The issue is mainly for keeping track of the idea. Hopefully, we find someone, maybe from the AsciiDoc community, to implement it...
